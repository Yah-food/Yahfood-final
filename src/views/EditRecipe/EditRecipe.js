import React, { useState, useEffect } from 'react';
import { useParams, useHistory } from 'react-router-dom';
import { getUserRecipeByTitle } from '../../services/fetchData';
import RecipeForm from '../../components/RecipeForm/RecipeForm';
import { editRecipe } from '../../services/users';
export default function EditRecipe() {
  const [recipe, setRecipe] = useState({});
  const params = useParams();
  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      const data = await getUserRecipeByTitle(params.title);
      setRecipe(data);
    };
    fetchData();
  }, [params.title]);

  const updateRecipeState = (key, value) => {
    recipe[key] = value;
    setRecipe({ ...recipe });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await editRecipe(recipe);
      alert('Recipe has been edited!');
    } catch {
      alert('Error, try again');
    }
    history.push(`/recipe/${recipe.title}`);
  };



  return (
    <div>
      <RecipeForm {...recipe} handleSubmit={handleSubmit} updateRecipeState={updateRecipeState} />
    </div>
  );
}
