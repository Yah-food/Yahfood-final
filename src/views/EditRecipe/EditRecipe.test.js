import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom/cjs/react-router-dom.min';
import EditRecipe from './EditRecipe';

test('renders edit individual recipe', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/profile/editrecipe/Birria']}>
      <Route path="/profile/editrecipe/:title" component={EditRecipe} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('Birria');

  expect(container).toMatchSnapshot();
});
