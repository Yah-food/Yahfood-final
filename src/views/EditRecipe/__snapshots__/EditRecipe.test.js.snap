// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders edit individual recipe 1`] = `
<div>
  <div>
    <div
      class="pt-2"
    >
      <a
        class="mx-5"
        href="/profile"
      >
        <button>
          Back
        </button>
      </a>
    </div>
    <div
      class="flex justify-center font-sans text-5xl opacity-70 py-10"
    >
      <h1>
        yahfood!
      </h1>
    </div>
    <div>
      <form
        class="py-20"
      >
        <label
          class="flex justify-center items-center flex-col"
        >
          Title:
          <input
            class=" border-solid rounded-lg border-2 w-40"
            name="title"
            type="text"
            value="Birria"
          />
        </label>
        <label
          class="flex justify-center items-center flex-col"
        >
          Ingredients:
          <textarea
            class=" border-solid rounded-lg border-2 w-40"
            name="description"
            type="text-area"
          >
            Meat (Lamb), dried chili peppers
          </textarea>
        </label>
        <label
          class="flex justify-center items-center flex-col"
        >
          Instructions:
          <textarea
            class=" border-solid rounded-lg border-2 w-40"
            name="instructions"
            type="text-area"
          >
            It is a traditional ancestral soup or stew made from a combination of chili pepper-based goat meat adobo, garlic, cumin, bay leaves, and thyme, and cooked at a low heat.
          </textarea>
        </label>
        <label
          class="flex justify-center items-center flex-col"
        >
          Image url:
          <input
            class=" border-solid rounded-lg border-2 w-40"
            name="images"
            type="text"
            value="https://www.thespruceeats.com/thmb/g1NmB3Jzsoe9TlGbEKRN1mNvMJg=/1500x1000/filters:fill(auto,1)/birria-tacos-recipe-5189284-hero-01-56190b7eb77b4370a0bf7f2341e94ee4.jpg"
          />
        </label>
        <div
          class="flex justify-center mx-auto my-5 w-64 items-center border-2 border-lime-700/50 rounded-lg bg-lime-700/50 "
        >
          <button>
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
